internal/uploader/events_uploader_handler.go

```diff
@@ -0,0 +1,247 @@
+package uploader
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"net/textproto"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/openshift/assisted-service/internal/common"
+	eventsapi "github.com/openshift/assisted-service/internal/events/api"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/k8sclient"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+var _ UploadHandler = &EventsUploader{}
+
+type EventsUploader struct {
+	Config
+	db     *gorm.DB
+	log    log.FieldLogger
+	client k8sclient.K8SClient
+}
+
+func NewEventsUploader(cfg *Config, db *gorm.DB, log logrus.FieldLogger, client k8sclient.K8SClient) *EventsUploader {
```

Comment:
Is anyone using this directly? I figure all callers would be using the function that returns an interface.

Reply:
Good point, removed this function!